### PR TITLE
More Winter Coats - Expansion Pack

### PR DESCRIPTION
**Pull request by Nakalon**
## About The Pull Request

This PR adds more Winter Coats to the game.
Because Winter Coats are cute, dressing up is fun, and some of them needed a touch up.

The following were added:
image (23)
1 - Virologist
2 - Geneticist
3 - Chemist
4 - Cargo
5 - Mining
6 - Brig Physician
7 - Detective

Within medbay, Viro, Geneticist and chemist, I took the armor values of the medbay coat and allocated them in a way that makes sense to me. They have colours relating to their job. They are available in their respective vendors.

Virologist has more more bio resistance, less of acid.
Geneticist has less bio and more rad. (The geneticist design has the features of a medcoat and the stripe of a science coat, because genetics is like a science)
Chemist has no bio but has Fire and a bit of bomb.
Chemist winter coat is also flame and acid-resistant. (This is not really a balance choice, more of a flavor one. These only really affect the item)

Cargo's Coat had its sprite modernized. It was painted in the colours of the old coat.
The Old Cargo coat can be bough on the hacked cargo vendor.
Cargo's Coat now accepts items that are relevant to the job of cargo (like the export scanner, clipboard, and drink bottles because cargo are hardy men?)

Mining Coat had its sprite modernized. It resembles the exploration suit miners use.
The Old coat was already mostly unobtainable and since its ugly I just kept it that way.
This was made to have similar functionality to the exploration suit, it is meant to be used in real mining work.
It has almost the same values but less bio and rad protection.
It can be upgraded with goliath plates all the same, and it can store miner related items in its slot:
Pickaxes, PKA, Resonator, Scanner, and again, a DRINK BOTTLE.
It is available in the miner equipment lockers.

The Brig Physician Coat bears a resemblance to the vest.
It has less armor values than the vest, but a tiny bit of laser to account for friendly fire incidents.
Its available on the brig phys locker.

The Detective Coat was also made to resemble the detective's jacket.
It has less armour values than the jacket, overall.
It fits all the items important for Detective activities such as the Peperspray, scanner, wooden baton, gun, etc.

## Why It's Good For The Game

This gives a tiny bit of love to some jobs that don't get many new things to cherish.
I can't exactly explain how much I like winter coats, and I love the colours of viro and chemist specially, so I really thought this would add a good amount of colour and choice for players.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image (23)
image (24)
image (25)
image (26)
image (27)
image

</details>

## Changelog
:cl: PinkSuzuky
add: Added 5 Winter Coats, Virologist, Geneticist, Chemist, Brig Physician and Detective. Made them available on the relevant vendors, with the exception of the Brig Physician Coat which can be acquired from the Brig Physician Locker.
imageadd: Updated the sprites of the Miner Winter Coat and Cargo Winter coat.
tweak: Changed some values on the Miner Winter Coat to make usable for mining.
tweak: Made the Miner Winter Coat available inside the mining equipment lockers.
tweak: Cargo Winter Coat now accepts Cargo related items in its item slot.
code: reorganized some lists on wintercoat code for readeability.
/:cl:

### CODE REVIEW

File: code/modules/clothing/suits/miscellaneous.dm [hunk 1]
```diff
@@ -634,6 +641,57 @@
...
+	name = "detective winter coat"
+	icon_state = "coatdetective"
+	item_state = "coatdetective"
+	allowed = list(
+		/obj/item/tank/internals,
+		/obj/item/melee/classic_baton,
+		/obj/item/gun/ballistic/revolver/detective,
+		/obj/item/detective_scanner,
+		/obj/item/flashlight,
+		/obj/item/taperecorder
```
Comment (Tsar-Salat) [inline code/modules/clothing/suits/miscellaneous.dm]:
```suggestion
		/obj/item/taperecorder,
```

Comment (Nakalon) [reply code/modules/clothing/suits/miscellaneous.dm]: guh my bad, I'll hold on on fixing this, I thinking I misinterpreted how the lists on subtypes would interact with the original ones.
Are they additive or do they replace completely the parent list?

Comment (Tsar-Salat) [reply code/modules/clothing/suits/miscellaneous.dm]: Adding a list to the subtype redefines it. It will not inherit the parents list if you redefine it.

Comment (Nakalon) [reply code/modules/clothing/suits/miscellaneous.dm]: Then I'll have to redo it tomorrow ACH, thanks...

**Comment by Harranhall** [comment]
So cosy!

**Comment by Nakalon** [comment]
I missed what I messed up this should now be extra safe.

**Review by PestoVerde322 — APPROVED** [review]

File: code/modules/clothing/suits/miscellaneous.dm [hunk 2]
```diff
@@ -713,20 +898,71 @@
...
+	armor = list(MELEE = 30,  BULLET = 20, LASER = 20, ENERGY = 20, BOMB = 50, BIO = 10, RAD = 20, FIRE = 50, ACID = 50, STAMINA = 20)
+	resistance_flags = FIRE_PROOF
+	high_pressure_multiplier = 0.4
 	hoodtype = /obj/item/clothing/head/hooded/winterhood/miner
 
+/obj/item/clothing/suit/hooded/explorer/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/armor_plate)
+
+/obj/item/clothing/head/hooded/explorer/Initialize(mapload)
```
Comment (itsmeow) [inline code/modules/clothing/suits/miscellaneous.dm]: is this meant to be for the miner coat?

Comment (Nakalon) [reply code/modules/clothing/suits/miscellaneous.dm]: It was! I coded this part at 4 am, guh, its fixed now!!!

**Review by itsmeow — APPROVED** [review]

**Pull request merged**